src/gtc/unstructured/usid_codegen.py

```diff
@@ -354,7 +354,7 @@ class UsidNaiveCodeGenerator(UsidCodeGenerator):
             for(std::size_t idx = 0; idx < gridtools::next::connectivity::size(${ prim_conn.name }); idx++) {
                 % if len(prim_sid.entries) > 0:
                 auto ${ prim_sid.ptr_name } = ${ prim_sid.origin_name }();
-                gridtools::sid::shift(${ prim_sid.ptr_name }, gridtools::device::at_key<
+                gridtools::sid::shift(${ prim_sid.ptr_name }, gridtools::host_device::at_key<
```

Comment:
Here the computation is always on host. can be just `gridtools::at_key`

Reply:
Same reasoning as above (for the CUDA version of the kernel), I'd keep it for now.